types/test/sortByProps.ts

```diff
@@ -15,12 +15,19 @@ const clara = {
   age: 314.159,
   lastName: 'Kris',
 };
-const people = [clara, bob, alice];
+const immutablePeople = [clara, bob, alice] as const;
+const mutablePeople = [clara, bob];
+const immutableProps = ['lastName', 'name'] as const;
+const mutableProps = ['age'];
 
-RA.sortByProps(['name'])(people); // $ExpectType object[]
-RA.sortByProps(['p'])(people); // $ExpectType object[]
-RA.sortByProps(['lastName', 'name'])(people); // $ExpectType object[]
-RA.sortByProps(['lastName', 'p', 'name'])(people); // $ExpectType object[]
+// tslint:disable:max-line-length
+RA.sortByProps(mutableProps)(immutablePeople); // $ExpectType ({ name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; })[]
+RA.sortByProps(['p'])(immutablePeople); // $ExpectType ({ name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; })[]
+RA.sortByProps(immutableProps)(immutablePeople); // $ExpectType ({ name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; } | { name: string; age: number; lastName: string; })[]
+RA.sortByProps(immutableProps)(mutablePeople); // $ExpectType { name: string; age: number; lastName: string; }[]
```

Comment:
I don't get this at all, can you pls explain the changes?

Reply:
Of course, here we go:

Currently, `sortByProps` accepts any `string` inside array parameter, so even if your array is composed by object with properties say x and y, you could pass `['z']` and returns an `object[]`, which is not well typed, because you can't get autocompletion with this even if your input is typed. Now this method strict it to receive only valid properties and return the same type as received. Hope this clarifies :)

Reply:
Yeah I get it.